components/prize_pool/commons/prize_pool_import.lua

```diff
@@ -0,0 +1,504 @@
+---
+-- @Liquipedia
+-- wiki=commons
+-- page=Module:PrizePool/Import
+--
+-- Please see https://github.com/Liquipedia/Lua-Modules to contribute
+--
+
+local Lua = require('Module:Lua')
+local Array = require('Module:Array')
+local ArrayExt = require('Module:Array/Ext')
+local DateExt = require('Module:Date/Ext')
+local Logic = require('Module:Logic')
+local MathUtil = require('Module:MathUtil')
+local String = require('Module:StringUtils')
+local Table = require('Module:Table')
+
+local MatchGroupCoordinates = Lua.import('Module:MatchGroup/Coordinates', {requireDevIfEnabled = true})
+local MatchGroupUtil = Lua.import('Module:MatchGroup/Util', {requireDevIfEnabled = true})
+local Opponent = Lua.import('Module:Opponent', {requireDevIfEnabled = true})
+local Placement = Lua.import('Module:PrizePool/Placement', {requireDevIfEnabled = true})
+local TournamentUtil = Lua.import('Module:Tournament/Util', {requireDevIfEnabled = true})
+
+local GROUPSCORE_DELIMITER = '-'
+local SCORE_STATUS = 'S'
+local DEFAULT_ELIMINATION_STATUS = 'down'
+local THIRD_PLACE_MATCH_ID = 'RxMTP'
+local GSL_GROUP_OPPONENT_NUMBER = 4
+local SWISS_GROUP_TYPE = 'swiss'
+local GSL_STYLE_SCORES = {
+	{2, 0, 0},
+	{2, 0, 1},
+	{1, 0, 2},
+	{0, 0, 2},
+}
+
+local Import = {}
+
+function Import.run(placements, args)
+	Import.config = Import._getConfig(args, placements)
+
+	if Import.config.importLimit == 0 or not Import.config.matchGroupsSpec then
+		return placements
+	end
+
+	return Import._importPlacements(placements)
+end
+
+function Import._getConfig(args, placements)
+	if String.isEmpty(args.matchGroupId1) and String.isEmpty(args.tournament1)
+		and not Import._enableImport(args.import, args.importEnableStartDate) then
+
+		return {}
+	end
+
+	return {
+		importLimit = Import._importLimit(args.importLimit, placements),
+		matchGroupsSpec = TournamentUtil.readMatchGroupsSpec(args)
+			or TournamentUtil.currentPageSpec(),
+		groupElimStatuses = Table.mapValues(
+			mw.text.split(args.groupElimStatuses or DEFAULT_ELIMINATION_STATUS, ','),
+			mw.text.trim
+		),
+		groupScoreDelimiter = args.groupScoreDelimiter or GROUPSCORE_DELIMITER,
+		allGroupsUseWdl = Logic.readBool(args.allGroupsUseWdl),
+	}
+end
+
+function Import._enableImport(importInput, importEnableStartDate)
+	local date = TournamentUtil.getContextualDate()
+	return Logic.nilOr(
+		Logic.readBoolOrNil(importInput),
+		importEnableStartDate and (not date or date >= importEnableStartDate)
```

Comment:
it is a date from which onwards import is enabled by default (e.g. for sc/sc2: 2022-01-14)
open for better names of course

Reply:
How about default on when called from new PPT while default off for legacy? 
With the args to activate obviously being being possible to turn on the legacy custom?

Reply:
could do the date stuff in legacy for sc/sc2 and default it to on in new one

for other wikis that already use the standardized module it could be an issue though

Reply:
What would the impact be?

Reply:
if the ppt only displays until place X it would still import all available on the page

if the stages are not specified correctly and the page has subpages or convoluted stages (bracket into groups into bracket etc) it will possibly return bs

what we could do is define a global start date for the new PPTs using import
sc/sc2 could then just overwrite via /Custom

Reply:
was thinking of this:
https://liquipedia.net/commons/index.php?title=Module%3APrizePool%2FImport&type=revision&diff=451677&oldid=451674 (date up for discussion)
https://liquipedia.net/commons/index.php?title=Module%3APrizePool%2FStarcraft&type=revision&diff=451678&oldid=449199
https://liquipedia.net/commons/index.php?title=Module%3APrizePool%2FLegacy%2FStarcraft&type=revision&diff=451679&oldid=449198

(already tested)